src/Components/ErrorPages/BrowserWarning.tsx

```diff
@@ -0,0 +1,42 @@
+import  { useEffect, useState } from "react";
+import browserslist from "browserslist";
+import { getUserAgentRegex } from "browserslist-useragent-regexp";
+import packageJson from "../../../package.json";
```

Comment:
Is it possible to achieve this without importing `package.json`?

Reply:
Let me check please

Reply:
Actually as per my understanding, this is not possible otherwise we need to pass hardcore data in the func called browerlist which is generating regex for supported browsers like this : 
`const supportedBrowsers = browserslist(">0.2%, not dead, not op_mini all");`
I would like to know any suggesting regarding the same

Reply:
the readme has an alternate method: https://github.com/browserslist/browserslist-useragent-regexp

Reply:
We need to use the `browserslist` config from `package.json` to generate the regex for supported browsers, rather than creating a separate `.browserslistrc` file. This ensures future changes to the `browserslist` config in `package.json` will automatically be reflected, making the process easier and maintaining consistency.

Reply:
According to the docs at: [https://github.com/browserslist/browserslist-useragent-regexp](https://github.com/browserslist/browserslist-useragent-regexp) , you can precompile the regex from the either the `package.json` or the `.browserslistrc` file, I have tested this out on a test project locally, browserslist-useragent-regexp will fetch the key `browserslist` from the `package.json`. And as for maintaining consistency between the regexp and the key in the `package.json`, can we introduce a new script into the build flow so that the regexp is recompiled every time a new build is made?

Also this will provide a bit more optimization, because a lot of redundant recompilation of the regexp will be reduced.

Reply:
yup! add the script as mentioned in the readme

Reply:
@AdityaJ2305 Should I make the required changes or would you be doing the same?

Reply:
Hey @itsRaCl , great suggestion. Yes, you can make the changes

Reply:
As suggested by @itsRaCl, we need to modify the build script to include a command that generates the updated regex during the build process. Like following way: 
`"build": "npm run generate-build-meta && npm run supported-browsers && npm run build:react"`
previously 
`"build": "npm run generate-build-meta && npm run build:react"`
 Is this modification permissible?"